internal/api/errors.go

```
@@ -81,16 +121,27 @@ func requestIDFrom(ctx context.Context) string {
 // Non-apiError values are internal faults: logged, reported as api_error
 // without leaking internals.
 func writeError(w http.ResponseWriter, r *http.Request, err error) {
+	inner := map[string]string{}
+	if withFields, ok := err.(*apiErrorWithFields); ok {
```

golangci-lint (2.12.2)

[error] type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors

(errorlint)